File: Modul-6/Soal-4/PRAK604-2410817210027-HafizPerdana.c
#include <stdio.h>
#include <string.h>

int main() {
char kode[100],pesan[100];
int bintang=0,pagar=0;

fgets(kode, sizeof(kode), stdin);
fgets(pesan, sizeof(pesan), stdin);

kode[strcspn(kode, "\n")]= '\0';
pesan[strcspn(pesan, "\n")]= '\0';

if(strlen(kode) != strlen(pesan)) {
    printf("\nPanjang Kalimat berbeda, Pesan Palsu");
    return 0; }

printf("\n");
int panjang=strlen(kode);
for(int i=0; i<panjang; i++){
    if(kode[i]==' ' && pesan[i]==' '){
        continue;
    }

    if(kode[i]==pesan[i]) {
        printf("*");
        bintang++;}
        
    else { 
        printf("#");
    pagar++;}
}

printf("\n*=%d",bintang);
printf("\n#=%d",pagar);

if(bintang>=pagar) {
    printf("\nPesan Asli");
    }
else {
    printf("\nPesan Palsu");
}

return 0;
}

File: Modul-6/Soal-4/PRAK604-2410817210027-HafizPerdana.py
kode = input().strip()
pesan = input().strip()

bintang = 0
pagar = 0

if len(kode) != len(pesan):
    print("\nPanjang Kalimat berbeda, Pesan Palsu")
else:
    print("")
    panjang = len(kode)
    for i in range(panjang):
        if kode[i] == ' ' and pesan[i] == ' ':
            continue

        if kode[i] == pesan[i]:
            print("*", end="")
            bintang += 1
        else:
            print("#", end="")
            pagar += 1

    print("\n*={}".format(bintang))
    print("#={}".format(pagar))

    if bintang >= pagar:
        print("Pesan Asli")
    else:
        print("Pesan Palsu")